### ISSUE TITLE
Problems building

### ISSUE THREAD

**Issue by charJe** [visualboyadvance-m/visualboyadvance-m#868]
### Expected behavior
compile successfuly

### Actual behavior
error invoking cmake 
### Steps to reproduce the behavior

```sh
$ guix build -f visualboy.scm
```
It appears to be a problem related to pthread, but nothing online has been helpful. I can compile gcc-ar-wrap.c just fine with `gcc -pthread gcc-ar-wrap.c`.

[compilation-log.txt](https://github.com/visualboyadvance-m/visualboyadvance-m/files/6986697/compilation-log.txt)
[visualboy.scm.txt](https://github.com/visualboyadvance-m/visualboyadvance-m/files/6986708/visualboy.scm.txt)
[CMakeError.log](https://github.com/visualboyadvance-m/visualboyadvance-m/files/6986710/CMakeError.log)
[CMakeOutput.log](https://github.com/visualboyadvance-m/visualboyadvance-m/files/6986711/CMakeOutput.log)

### Build details

Operating System (Windows, Mac, Linux (state distribution), etc.): GNU Guix

If using Linux, specify if using xorg or Wayland: xorg

Version of code (pre-built binary version, or commit ref, or just "master"):  v2.1.4

**Comment by ZachBacon** [comment]
Not to sound harsh, but this isn't our bug, this is an guix issue, this builds fine on other linux platforms.

**Comment by charJe** [comment]
Sure, it isn't a problem with your software, but you know the most about your software, its depenndencies, its required build environment. Thus you would be the most qualified to help me successfully build and bring vba to another system. I am mainly just wondering if you can see through the pthread issue; maybe I am missing some dependency (I have tried boost), or something else I am not aware of.

**Comment by ZachBacon** [comment]
Boost isn't a dependency, if you want to know the basic dependencies, look at the readme hosted on this github of what's required to build vba-m

**Comment by charJe** [comment]
I believe I have all the dependencies installed (as described in the `inputs` section of visualboy.scm). I'm not familiar with cmake, so I find the errors hard to parse through. This is especially the case since I read that this pthread issue can sometimes hide other, unrelated issues.

**Comment by ZachBacon** [comment]
I'll try, but I make no promises, but I'll try and investigate this, but again I'm very certain it's your distribution.

**Comment by charJe** [comment]
I was able to compile it, so it should be in guix repositories soon.
